### DISCUSSION TITLE
DISKANN索引问题

### DISCUSSION THREAD

**Discussion by deict** [milvus-io/milvus#33352] [Q&A and General discussion]
ubuntu20.04  python 3.8， milvus 2.4.1 cpu版本，使用的DISKANN索引类型，昨天夜里插入了1500万条数据，DIM：384，图片是加载内存并检索使用的内存情况，我想问下这个走磁盘索引了吗？

加载之前内存：
![2](https://github.com/milvus-io/milvus/assets/112455517/d72de615-b3e9-40f2-81a3-d72c5fbc9af0)

加载之后内存：
![1](https://github.com/milvus-io/milvus/assets/112455517/e6571b08-47db-4b07-88e7-b169da4413bf)

还有这个cpu占用这么高，这个正常吗？
<img width="914" alt="image" src="https://github.com/milvus-io/milvus/assets/112455517/8fca716f-19b2-48c3-9673-a5ce1032f9ea">

**Comment by yhmo** [comment]
打印出索引信息，如果是DISKANN那就是DISKANN。
打印出索引的进度，如果indexed_rows等于total_rows，那就是索引已经建好。
```
for index in collection.indexes:
    print(index.to_dict())
    
print(utility.index_building_progress(collection_name=collection_name))
```

DISKANN虽然名字叫磁盘索引，它也需要内存加载一部分数据，一般是数据量的1/4到1/6。它也需要CPU做密集的计算。
1500万条384维数据的数据量是21GB，那么DISKANN大约需要4到5GB的内存，上下浮动一些也有可能，内存用量从5.4G涨到13GB也算正常。
